BLSQ/iaso
# IA-3154 default source version and tree view

Tree view is not a set on selected version while loading
Remove datasources without versions from the dropdown options.
Preselect the latest version if no default version is available.
Display the description and indicate if a version is the default in the version dropdown options.

Related JIRA tickets : IA-3154

## Self proofreading checklist


## Changes

- branch tree view on params to get selected version
- filter data source without version
- preselect first version if no default
- adapt version label to display number, desc and is default

## How to test

You need multiple data source:
- one without version
- one with versions but no default
- one with versions and a default

Go to org unit page and play with those sources, check you don't have the source without version.
Double check tree usage and version label on other pages

## Print screen / video

![Screenshot 2024-06-28 at 12 07 47](https://github.com/BLSQ/iaso/assets/12494624/0836b487-943e-45c8-a7a5-b4e07dd0993a)
![Screenshot 2024-06-28 at 12 07 29](https://github.com/BLSQ/iaso/assets/12494624/105e7977-e595-49d7-b9c9-54b505366ab9)
![Screenshot 2024-06-28 at 12 07 21](https://github.com/BLSQ/iaso/assets/12494624/49c8aa6d-1c56-4dbe-9c03-9b6617930cde)
